src/applications/check-in/utils/appointment/index.js

```diff
@@ -41,6 +45,39 @@ const appointmentWasCanceled = appointments => {
   return Array.isArray(appointments) && appointments.some(statusIsCanceled);
 };
 
+/**
+ * Get the interval from now until the end of the next check-in window.
+ *
+ * @param {Array<Appointment>} appointments
+ *
+ * @returns {number} ms until the end of the next check-in window. (0 if no appointments are eligible for check-in)
+ */
+const intervalUntilNextAppointmentIneligibleForCheckin = appointments => {
+  let interval = 0;
+
+  const eligibleAppointments = appointments.filter(
+    appointment => appointment.eligibility === ELIGIBILITY.ELIGIBLE,
+  );
+
+  let checkInWindowEnds = eligibleAppointments.map(
+    appointment => appointment.checkInWindowEnd,
+  );
+
+  checkInWindowEnds.sort((a, b) => {
+    return parseISO(a) > parseISO(b);
+  });
+
+  checkInWindowEnds = checkInWindowEnds.filter(
```

Comment:
Extremely nitpicky but potentially we could run the filter before the sort so that we aren't sorting more elements than we need to?

Reply:
done, I'm all about the micro-optimizations